### PR TITLE
Add units to Quantity

### PR DESCRIPTION
**Pull request by sc-arecrow**
# TODO
* [x] Change Ingredient Models
* [x] Change `QuantityRecord`
* [x] Change existing tests
* [x] Write new tests
* [x] Solve bug where all dates become `nil` when passed into ingredient

### CODE REVIEW

File: ChopChop/Models/Ingredients/Ingredient/Ingredient.swift [hunk 1]
```diff
@@ -54,8 +54,20 @@ class Ingredient: FetchableRecord {
...
+
+            return expiryDate >= .today
+        }
+    }
+
+    var totalUsableQuantity: Double {
+        notExpiredBatches
+            .map { $0.quantity.baseValue }
+            .reduce(0.0, +)
     }
```
Comment (seowalex) [inline ChopChop/Models/Ingredients/Ingredient/Ingredient.swift:77]: Should we also include a `totalQuantity` computed variable?

File: ChopChop/Storage/Models/Ingredients/QuantityRecord.swift [hunk 2]
```diff
@@ -1,23 +1,25 @@
...
-        } else if let value = try container.decodeIfPresent(Double.self, forKey: .mass) {
-            self = .mass(value)
-        } else if let value = try container.decodeIfPresent(Double.self, forKey: .volume) {
-            self = .volume(value)
+        } else if let value = try container.decodeIfPresent(Double.self, forKey: .mass),
+                  let unit = try container.decodeIfPresent(MassUnit.self, forKey: .unit) {
+            self = .mass(value, unit: unit)
+        } else if let value = try container.decodeIfPresent(Double.self, forKey: .volume),
+                  let unit = try container.decodeIfPresent(VolumeUnit.self, forKey: .unit) {
+            self = .volume(value, unit: unit)
```
Comment (seowalex) [inline ChopChop/Storage/Models/Ingredients/QuantityRecord.swift:22]: Has this been confirmed to be working properly with the database (both the encoding and decoding)?

Comment (sc-arecrow) [reply ChopChop/Storage/Models/Ingredients/QuantityRecord.swift:22]: I think this requires some updating of the database to also store the unit of the quantity.

File: ChopChop/Models/Ingredients/Ingredient/Ingredient.swift (hunk 1 above)
Comment (sc-arecrow) [reply ChopChop/Models/Ingredients/Ingredient/Ingredient.swift:77]: Yes that makes sense

**Pull request merged**